packages/core/src/api/apis/definitions/featureFlags.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ApiRef from '../ApiRef';
+import { FeatureFlagName } from '../../plugin/types';
+
+/**
+ * The feature flags API is used to toggle functionality to users across plugins and Backstage.
+ *
+ * Plugins can use this API to register feature flags that they have available
+ * for users to enable/disable, and this API will centralize the current user's
+ * state of which feature flags they would like to enable.
+ *
+ * This is ideal for Backstage plugins, as well as your own App, to trial incomplete
+ * or unstable upcoming features. Although there will be a common interface for users
+ * to enable and disable feature flags, this API acts as another way to enable/disable.
+ */
+export type FeatureFlagsApi = {
+  /**
+   * Get the current user's status of a Feature Flag
+   *
+   * @returns bool True if the current user has enabled the feature flag
+   */
+  getItem(name: FeatureFlagName): boolean;
+
+  /**
+   * Enable an registered feature flag.
+   */
+  enable(name: FeatureFlagName): void;
```

Comment:
I'm inclined to go with `set(name: FeatureFlagName, enabled: boolean)` instead, along with `isEnabled(name: FeatureFlagName)`, or just `get(...)`. Reason for putting it in one method is just because fewer methods = yay.

Also though about using an enum instead of boolean for the state, since that's a bit more forwards compatible. Not convinced we need that though.

Reply:
Looking at its current implementation in internal, it seems like it's semi following the `window.localStorage` API with the `getItem` method although not so much with `enable`, `disable`, etc. As mentioned in other responses, mostly designed with backwards compatibility.

I like having a `setItem` or `set` method as it's more explicit and makes adding conditional logic better (i.e. `api.set('flag name', !!ifTrue)` vs `if (!!ifTrue) { api.set('flag name', true) } else { ... }`). Will update it accordingly.

On a side note, if these things are being renamed/rethought, probably worth starting to build up a migration doc for internal to external

Reply:
Agree on the migration doc. Since this is more or less a rename it's easy enough to bring in anyway, but there will definitely be some more intricate things that need documentation to be written. Then again we're aiming to get this repo to a point were we can use it internally asap, and once that's in place the migration will be a bit more unified with the design process in this repo.

For now I'd say freely change things that don't make sense in the internal implementation, that tiny bit of refactoring is definitely worth it imo.